File: java/log-levels/src/main/java/LogLevels.java
import java.util.Arrays;

public class LogLevels {

    public static String message(String logLine) {
        return logLine.split(":")[1].trim();
    }

    public static String logLevel(String logLine) {
        String level = logLine.split(":")[0];

        return level.substring(1, level.length() - 1).trim().toLowerCase();
    }

    public static String reformat(String logLine) {
        return String.format("%s (%s)", message(logLine), logLevel(logLine));
    }
}

File: fsharp/log-levels/LogLevels.fs
module LogLevels

let message (logLine: string) : string = logLine.Split(':').[1].Trim()

let logLevel (logLine: string) : string =
    logLine.Split(':').[0].Trim('[', ']').ToLower()

let reformat (logLine: string) : string =
    $"""{message logLine} ({logLevel logLine})"""
